wp/lib/bap_wp/src/z3_utils.ml

```diff
@@ -166,7 +166,8 @@ let asserts_of_model (model_string : string) (sym_names : string list) : Sexp.t
             model_val
           | Sexp.List _  ->  (* Function model *)
             Sexp.List [Sexp.Atom "lambda" ; args; model_val] (* Sexp.Atom varname *)
-          | Sexp.Atom _ -> failwith "Unexpected atom"
+          | Sexp.Atom a -> failwith (sprintf "Unexpected atom %s in external model %s"
```

Comment:
Do we want to print the whole model here? I wonder if it just drowns the user.

Reply:
We do. In the case this error message gets printed I need someone to go in and add print statements so I can see the model to know what went wrong.

Reply:
I guess I could put the error message after the model so it isn't as buried?

Reply:
All right.

Reply:
If we're forcing a fail with a message, then it's clearly not a problem for the average user. This event itself will "drown" the average user. For these kind of hard crashes that users can't really handle themselves, I think it's best to print the problem fully, so that the user who encounters it can just cut and paste the data into a github issue.